src/workspace.rs

```diff
@@ -22,45 +23,62 @@ impl Workspace {
         Ok(workspace)
     }
 
-    pub fn set_up_files(&self, job: &job::Job) -> Result<()> {
+    pub fn set_up_files(
+        &self,
+        job: &job::Job,
+        job_to_store_path: &HashMap<job::Key<job::Base>, store::Item>,
+    ) -> Result<()> {
         for file in &job.input_files {
-            if let Some(parent_base) = file.parent() {
-                let parent = self.join(parent_base);
-
-                if !parent.exists() {
-                    fs::create_dir_all(parent).with_context(|| {
-                        format!("could not create parent for `{}`", file.display())
-                    })?;
-                }
-            }
+            self.set_up_path(file, file)?
```

Comment:
We verbatim pass input files through a build to the output? That sounds odd. At least that is how I read this.

Reply:
hmm, I'm not sure what you mean here. Could you say more?

What's going on here: we're setting up the workspace files. The LHS of `set_up_path` is the path outside the workspace, and the RHS is the path inside it.

Reply:
Ah, that makes sense.  I was thinking this was setting up the exports of a build that other builds can consume.